.circleci/config.yml

```diff
@@ -0,0 +1,69 @@
+# CircleCI 2.0 configuration file. See <https://circleci.com/docs/2.0/language-python/>.
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:2.7-jessie
+    steps:
+      - checkout
+      - run:
+          name: Allow cache restore
+          command: |
+            sudo chown -R circleci:circleci /usr/local/bin
+            sudo chown -R circleci:circleci /usr/local/lib/python2.7/site-packages
+
+      - restore_cache:
+          environment:
+          key: cache-root-{{ checksum "setup.py" }}
+
+      - run:
+          name: Install execution dependencies
+          command: |
+            sudo pip install --upgrade pip wheel  # pip >= 8.0 needed to be compatible with "manylinux" wheels, used by numpy >= 1.11
+            # sudo pip install --editable git+https://github.com/openfisca/openfisca-core.git@BRANCH_NAME#egg=OpenFisca-Core  # use a specific branch of OpenFisca-Core
+            sudo pip install .[test] --upgrade
+
+      - run:
+          name: Install deployment dependencies
+          command: sudo pip install twine
+
+      - save_cache:
+          key: cache-root-{{ checksum "setup.py" }}
+          paths:
+            - /usr/local/lib/python2.7/site-packages/
+            - /usr/local/bin
+
+      - run:
+          name: Run tests
+          command: make test
+
+      - run:
+          name: Check version number has been properly updated
+          command: |
+            git fetch
+            ./.circleci/check-version-bump.sh
+
+  deploy:
```

Comment:
Have you really tried this ? 

It seems like CircleCI doesn't persist anything from one job to another, unless you use [workflows or cache](https://circleci.com/blog/persisting-data-in-workflows-when-to-use-caching-artifacts-and-workspaces/). 

So when running this job, I'd expect the current folder not to be checked out, and the dependencies not being installed.

Reply:
I did only partially: the syntax was correct but I was missing a username and password to actually test the deployment. I agree with your assumptions of the build folder being empty.